crates/ibc/src/core/ics04_channel/events.rs

```diff
@@ -1,719 +1,733 @@
+    acknowledgement: AcknowledgementAttribute,
+    dst_connection_id: PacketConnectionIdAttribute,
 }
 
 impl WriteAcknowledgement {
-    pub fn src_port_id(&self) -> &PortId {
-        &self.packet.source_port
-    }
-    pub fn src_channel_id(&self) -> &ChannelId {
-        &self.packet.source_channel
-    }
-    pub fn dst_port_id(&self) -> &PortId {
-        &self.packet.destination_port
-    }
-    pub fn dst_channel_id(&self) -> &ChannelId {
-        &self.packet.destination_channel
-    }
-}
-
-impl From<WriteAcknowledgement> for IbcEvent {
-    fn from(v: WriteAcknowledgement) -> Self {
-        IbcEvent::WriteAcknowledgement(v)
+    pub fn new(
+        packet: Packet,
+        channel_ordering: Order,
+        dst_connection_id: ConnectionId,
+        acknowledgement: Acknowledgement,
+    ) -> Self {
+        Self {
+            packet_data: packet.data.into(),
+            timeout_height: packet.timeout_height.into(),
+            timeout_timestamp: packet.timeout_timestamp.into(),
+            sequence: packet.sequence.into(),
+            src_port_id: packet.source_port.into(),
+            src_channel_id: packet.source_channel.into(),
+            dst_port_id: packet.destination_port.into(),
+            dst_channel_id: packet.destination_channel.into(),
+            channel_ordering: channel_ordering.into(),
+            acknowledgement: acknowledgement.into(),
+            dst_connection_id: dst_connection_id.into(),
+        }
     }
 }
 
 impl TryFrom<WriteAcknowledgement> for AbciEvent {
     type Error = Error;
 
     fn try_from(v: WriteAcknowledgement) -> Result<Self, Self::Error> {
-        let mut attributes = Vec::<Tag>::try_from(v.packet)?;
-        let val =
-            String::from_utf8(v.ack).expect("hex-encoded string should always be valid UTF-8");
-        // No actual conversion from string to `Tag::Key` or `Tag::Value`
-        let ack = Tag {
-            key: PKT_ACK_ATTRIBUTE_KEY.parse().unwrap(),
-            value: val.parse().unwrap(),
-        };
-        attributes.push(ack);
+        let mut attributes = Vec::with_capacity(11);
+        attributes.append(&mut v.packet_data.try_into()?);
+        attributes.push(v.timeout_height.into());
+        attributes.push(v.timeout_timestamp.into());
+        attributes.push(v.sequence.into());
+        attributes.push(v.src_port_id.into());
+        attributes.push(v.src_channel_id.into());
+        attributes.push(v.dst_port_id.into());
+        attributes.push(v.dst_channel_id.into());
+        attributes.push(v.channel_ordering.into());
+        attributes.append(&mut v.acknowledgement.try_into()?);
+        attributes.push(v.dst_connection_id.into());
+
         Ok(AbciEvent {
             type_str: IbcEventType::WriteAck.as_str().to_string(),
             attributes,
         })
     }
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
+#[derive(Debug)]
 pub struct AcknowledgePacket {
-    pub packet: Packet,
+    packet_data: PacketDataAttribute,
```

Comment:
Seems like ibc-go added `packet_data` to the `AcknowledgePacket` event very recently so it's not in a released version yet. I think it would be easier if we targetted the latest release (i.e. v5.0.1) instead of `main` because it makes it easier to track changes. WDYT?

Reply:
[ee3b479](https://github.com/cosmos/ibc-rs/pull/198/commits/ee3b479b88cd1db273d06c8becc3bb35d716b2d6)